dubbo-common/src/main/java/org/apache/dubbo/rpc/service/GenericService.java

```diff
@@ -37,12 +37,24 @@
      */
     Object $invoke(String method, String[] parameterTypes, Object[] args) throws GenericException;
 
+    /**
+     * Call target method on the consumer side asynchronously, the provider side should not be called the $invokeAsync method.
+     *
+     * @param method         Method name, e.g. findPerson. If there are overridden methods, parameter info is
+     *                       required, e.g. findPerson(java.lang.String)
+     * @param parameterTypes Parameter types
+     * @param args           Arguments
+     * @return invocation return value
+     * @throws GenericException potential exception thrown from the invocation
+     */
     default CompletableFuture<Object> $invokeAsync(String method, String[] parameterTypes, Object[] args) throws GenericException {
-        Object object = $invoke(method, parameterTypes, args);
-        if (object instanceof CompletableFuture) {
-            return (CompletableFuture<Object>) object;
-        }
-        return CompletableFuture.completedFuture(object);
+        return CompletableFuture.supplyAsync(() -> {
+            Object object = $invoke(method, parameterTypes, args);
+            // if (object instanceof CompletableFuture) {
+            //     return ((CompletableFuture) object).get();
+            // }
+            return object;
+        });
```

Comment:
The initial motivation of introducing $invokeAsync is to support the direct calling of methods with the signature of CompletableFuture. 

This pr seems like a breaking API change for the CompletableFuture instance returned has been changed.

How about simply adding the following line to make $invoke asynchronously.
```java
RpcContext.getContext().setAttachment(ASYNC_KEY, "true");
```

Reply:
> The initial motivation of introducing $invokeAsync is to support the direct calling of methods with the signature of CompletableFuture.

Could the $invoke method return a CompletableFuture object?
If `$invoke()` is a synchronous call, calling directly here will block the thread, so I want to change it to asynchronous execution, by calling `$invoke()` in `CompletableFuture.supplyAsync()`.

Reply:
> Could the $invoke method return a CompletableFuture object?

Yes, only if the method signature is like:
```java
class Service {
     CompletableFuture<String> sayHello(); 
}
```

Reply:
> If $invoke() is a synchronous call, calling directly here will block the thread,

Yes, you are right, the code before will block the thread. I think adding an ASYNC_KEY attachment would solve this issue while avoiding the previous CompletableFuture instance issue. 

`RpcContext.getContext().setAttachment(ASYNC_KEY, "true");`

Reply:
> Yes, only if the method signature is like:
> class Service {
>      CompletableFuture<String> sayHello(); 
> }

This code should be the implementation code of the Service. The returned CompletableFuture object will be processed during serialization and will not be received on the consumer side. Not sure how the injvm protocol handles this situation.

Reply:
>  The returned CompletableFuture object will be processed during serialization and will not be received on the consumer side. 

Actually the CompletableFuture is not serialized and transmitted on the wire, it is disassembled at the consumer end and reassembled at the consumer side. Anyway, the completable future instance should not be changed.

Reply:
Sorry, I think I missed something here.
```java
if (object instanceof CompletableFuture) {
      return ((CompletableFuture) object).get();
  }
```